pkg/ds/daemon_set.go

```diff
@@ -135,17 +119,26 @@ func (ds *daemonSet) WatchDesires(
 				select {
 				case errCh <- err:
 				case <-quitCh:
+					return
 				}
 			}
 
 			select {
-			case newDS := <-updatedCh:
+			case newDS, ok := <-updatedCh:
+				if !ok {
+					// channel closed
+					return
+				}
 				ds.logger.NoFields().Infof("Received daemon set update signal: %v", newDS)
+				if newDS == nil {
+					ds.logger.NoFields().Fatal("Unexpected a nil daemon set during update", *ds)
```

Comment:
can I just ask whether it's OK for this to be fatal? Who will be running it and is it OK with stopping the entire process?